File: app/src/main/java/com/joyhonest/sports_camera/AppPrefs.java
package com.joyhonest.sports_camera;

import android.content.Context;
import android.content.SharedPreferences;

public class AppPrefs {
    private static final boolean DEFAULT_PRIVATE_POLICY_AGREEMENT = false;
    private static final String PREF_NAME = "Sports_Camera.pref";
    private static final String PREF_PRIVACY_POLICY_AGREEMENT = "perf_privacy_policy_agreement";
    private static AppPrefs instance;
    private final SharedPreferences prefs;

    private AppPrefs(Context context) {
        this.prefs = context.getSharedPreferences(PREF_NAME, 0);
    }

    public static synchronized AppPrefs getInstance(Context context) {
        AppPrefs appPrefs;
        synchronized (AppPrefs.class) {
            if (instance == null) {
                instance = new AppPrefs(context);
            }
            appPrefs = instance;
        }
        return appPrefs;
    }

    public boolean getPrivacyPolicyAgreement() {
        return this.prefs.getBoolean(PREF_PRIVACY_POLICY_AGREEMENT, false);
    }

    public void setPrivacyPolicyAgreement(boolean z) {
        this.prefs.edit().putBoolean(PREF_PRIVACY_POLICY_AGREEMENT, z).apply();
    }
}

File: app/src/main/java/com/joyhonest/sports_camera/AppPrefs.kt
package com.joyhonest.sports_camera

import android.content.Context
import android.content.SharedPreferences
import kotlin.jvm.Synchronized

class AppPrefs private constructor(context: Context) {
    private val prefs: SharedPreferences

    init {
        prefs = context.getSharedPreferences(PREF_NAME, 0)
    }

    var privacyPolicyAgreement: Boolean
        get() = prefs.getBoolean(PREF_PRIVACY_POLICY_AGREEMENT, false)
        set(z) {
            prefs.edit().putBoolean(PREF_PRIVACY_POLICY_AGREEMENT, z).apply()
        }

    companion object {
        private const val DEFAULT_PRIVATE_POLICY_AGREEMENT = false
        private const val PREF_NAME = "Sports_Camera.pref"
        private const val PREF_PRIVACY_POLICY_AGREEMENT = "perf_privacy_policy_agreement"
        private var instance: AppPrefs? = null

        @Synchronized
        fun getInstance(context: Context): AppPrefs {
            var appPrefs: AppPrefs
            synchronized(AppPrefs::class.java) {
                if (instance == null) {
                    instance = AppPrefs(context)
                }
                appPrefs = instance!!
            }
            return appPrefs
        }
    }
}